GVFS/GVFS.Common/GVFSPlatform.cs

```diff
@@ -39,10 +39,13 @@ public static void Register(GVFSPlatform platform)
         public abstract void StartBackgroundProcess(ITracer tracer, string programName, string[] args);
         public abstract bool IsProcessActive(int processId);
         public abstract void IsServiceInstalledAndRunning(string name, out bool installed, out bool running);
+        public abstract bool ServiceSupportsDriverLoading();
```

Comment:
Two suggestions:

1. Move `ServiceSupportsDriverLoading ` into the `IKernelDriver`
2. When adding methods to `GVFSPlatform` try to write methods that describe the behavior of the platform rather than the behavior of VFS4G on the platform.  So, rather than `ServiceSupportsDriverLoading` maybe go with a name like `IsDriverAutomaticallyLoaded` or `DoesDriverLoadAtBoot` and have it return `true` on the Mac platform and `false` on Windows.

Reply:
Doesn't the windows driver when inboxed load at boot automatically or is it still the service that is loading it?  I'm not sure this flag is a feature of the driver.  If we ran a sudo service on Mac that could kext load the driver wouldn't the service on Mac be able to support loading the kext if it wasn't loaded for some reason?

Reply:
> Doesn't the windows driver when inboxed load at boot automatically or is it still the service that is loading it?

Good point, I was a bit sloppy with terminology in my suggestion.

`TryEnableAndAttachPrjFltThroughService` was first added because on Windows (unlike Mac) ProjFS must be attached to each volume where ProjFS will be used (even today with inbox ProjFS).  Pre-RS4 versions of Windows require elevation to attach ProjFS which is why the mount verb sometimes needs to ask GVFS.Service to attach the filter.  From that perspective a more accurate name for this new method might be something like `IsDriverAttachedPerVolume`.

Since then, however, the `TryEnableAndAttachPrjFltThroughService` request has also grown in scope to include checking that ProjFS is properly installed, enabled, etc.  As you pointed out this functionality is something we might want on Mac as well.

It seems like we have a few options:

- I we do plan on having a version of `TryEnableAndAttachPrjFltThroughService` that works on Mac, then we should have a new `UnderConstruction` flag rather than a new method (this is the approach we've been using for planned features that aren't supported on Mac yet)

- Alternatively, if we plan to support this on Mac in the future we could call `TryEnableAndAttachPrjFltThroughService` on both platforms now and have it no-op in the Mac GVFS.Service (and I like your suggestion of picking a new name for the method).

- If we don't think we'll ever support an equivalent of `TryEnableAndAttachPrjFltThroughService` on Mac then I think this check could be moved to `IKernelDriver` because it sounds like some aspect of the platform (i.e. the need to attach the filter to specific volumes) would be driving the decision to call `TryEnableAndAttachPrjFltThroughService`

What do you think?

Reply:
From what I understand we will not support this on Mac because we are running the service per user and that is not likely to change.  Since that is the case today, sounds like the `IKernelDriver` is the place for it.  What are you thoughts on moving this check into the `IsReady` method since it looks like this is part of getting the driver "ready"?

Reply:
> From what I understand we will not support this on Mac because we are running the service per user and that is not likely to change.

There might be multiple reasons 😄 

I was under the impression we would not support this on Mac because there is no "attach filter to volume" step on Mac.

@alameenshah, are there plans to support this feature on Mac?  (and if not, why not?)

> What are you thoughts on moving this check into the IsReady method since it looks like this is part of getting the driver "ready"?

I think the benefit is that it will allow us to remove `ServiceSupportsDriverLoading`.  

The potential downside is that code for talking to GVFS.Service would be living in `ProjFSFilter` which feels a bit wrong to me.  The `ProjFSFilter` class doesn't know anything about VFS4G and its scope has been limited to interacting with ProjFS (and the parts of Windows that ProjFS depends on).

Reply:
One other downside is that we also call `TryEnableAndAttachPrjFltThroughService` in `ServiceVerb.Execute`, but `Execute` does not call `IsReady` and so assuming we don't want to ever call `TryEnableAndAttachPrjFltThroughService` on Mac we'd still need to address that somehow in the service verb.